google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/CommitterImpl.java

```diff
@@ -144,4 +151,13 @@ public void onClientResponse(SequencedCommitCursorResponse value) throws Checked
       return ApiFutures.immediateFailedFuture(e);
     }
   }
+
+  @Override
+  public void waitUntilEmpty() throws CheckedApiException {
+    try (CloseableMonitor.Hold h = monitor.enterWhenUninterruptibly(isEmptyOrShutdown)) {
+      if (shutdown) {
+        throw new CheckedApiException("Shutting down.", Code.UNAVAILABLE);
```

Comment:
this will cause clean shutdown to become unclean. If we're trying to shut down cleanly, the queue of acks being flushed to the server should become empty over time like normal. If we're failed, you should fail here with the error it failed with.

What you should do is 1) change hadPermanentError to Optional<CheckedApiException> permanentError, 2) use the existing isEmptyOrError condition to check and test permanentError.isPresent when determining if you're going to throw.

Reply:
Done